notification.c

```diff
@@ -286,3 +286,51 @@ void notification_handle_button(struct mako_notification *notif, uint32_t button
 		break;
 	}
 }
+
+/*
+ * Searches through the notifications list and returns the next position at which to insert.
+ * If no results for the specified urgency are found, it will return the closest link searching in the direction specifed. (-1 for lower, 1 or upper).
+ */
+static struct wl_list *get_last_notif_by_urgency(struct wl_list *notifications, enum mako_notification_urgency urgency, int direction) {
+	enum mako_notification_urgency current = urgency;
+
+	if (wl_list_length(notifications) == 0) {
+		return notifications;
+	}
+
+	while (current <= MAKO_NOTIFICATION_URGENCY_HIGH && current >= MAKO_NOTIFICATION_URGENCY_UNKNOWN) {
+		struct mako_notification *notif;
+		wl_list_for_each_reverse(notif, notifications, link) {
+			if (notif->urgency == current) {
+				return &notif->link;
+			}
+		}
+		current += direction;
+	}
+
+	return notifications;
+}
+
+void insert_notification(struct mako_state *state, struct mako_notification *notif) {
+	struct mako_config *config = &state->config;
+	struct wl_list *insert_node;
+
+	if (config->sort_criteria == MAKO_SORT_CRITERIA_TIME &&
+			!config->sort_asc) {
```

Comment:
`!config->sort_asc` should probably be `!(config->sort_asc & MAKO_SORT_ASC_TIME)`